tests/TestCase/Routing/Route/RouteTest.php

```diff
@@ -1247,6 +1248,99 @@ public function testParseWithPassDefaults()
         $this->assertEquals($expected, $result);
     }
 
+    /**
+     * Test parse with optional parameters at the beginning of the route
+     *
+     * @return void
+     */
+    public function testParseWithOptonalStartParameter()
+    {
+        // {start} may be omitted
+        $route = new Route(
+            '/{start}/{controller}',
+            ['start' => null],
+            ['start' => 'prefix|other']
+        );
+
+        // Test route against path with {pref}
+        $result = $route->parse('/prefix/posts', 'GET');
+        $expected = [
+            'start' => 'prefix',
+            'controller' => 'posts',
+            'pass' => [],
+            '_matchedRoute' => '/{start}/{controller}',
+        ];
+        $this->assertSame($expected, $result);
+
+        // Test route against path without {pref}
+        $result = $route->parse('/posts', 'GET');
+        $expected = [
+            'start' => null,
+            'controller' => 'posts',
+            'pass' => [],
+            '_matchedRoute' => '/{start}/{controller}',
+        ];
+
+        $this->assertSame($expected, $result);
+
+        // Test route against path with additional segment
+        // This should not match
+        $result = $route->parse('/posts/view', 'GET');
+        $this->assertNull($result);
+    }
+
+    /**
+     * Test parse with optional parameters at the end of the route
+     *
+     * @return void
+     */
+    public function testParseWithOptonalEndParameter()
+    {
+        // {end} may be omitted
+        $route = new Route(
+            '/{controller}/{end}',
+            ['end' => null],
+            ['end' => 'postfix|other']
+        );
+
+        // Test route against path with {end}
+        $result = $route->parse('/posts/postfix', 'GET');
+        $expected = [
+            'controller' => 'posts',
+            'end' => 'postfix',
+            'pass' => [],
+            '_matchedRoute' => '/{controller}/{end}',
+        ];
+        $this->assertSame($expected, $result);
+
+        // Test route against path without {end}
+        $result = $route->parse('/posts', 'GET');
+        $expected = [
+            'controller' => 'posts',
+            'pass' => [],
+            'end' => null,
+            '_matchedRoute' => '/{controller}/{end}',
+        ];
+
+        $this->assertSame($expected, $result);
+
+        // Test route against path without {pref}
+        $result = $route->parse('/posts/', 'GET');
+        $expected = [
+            'controller' => 'posts',
+            'pass' => [],
+            'end' => null,
+            '_matchedRoute' => '/{controller}/{end}',
+        ];
```

Comment:
This looks the same as the test on 1316-1323.

Reply:
The difference is the trailing '/' in parse in this case. The result should, of course, be the same.

Reply:
:+1: I missed the trailing slash.